tests/group_functions/group_broadcast.h

```diff
@@ -0,0 +1,213 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "group_functions_common.h"
+
+template <int D, typename T>
+class broadcast_group_kernel;
+
+/**
+ * @brief Provides test for broadcast values inside the group
+ * @tparam D Dimension to use for group instance
+ * @tparam T Type for broadcasted value
+ */
+template <int D, typename T>
+void broadcast_group(sycl::queue& queue) {
+  // 3 functions
+  constexpr int test_matrix = 3;
+  const std::string test_names[test_matrix] = {
+      "T group_broadcast(group g, T x)",
+      "T group_broadcast(group g, T x, group::linear_id_type local_linear_id)",
+      "T group_broadcast(group g, T x, group::id_type local_id)"};
+
+  sycl::range<D> work_group_range = util::work_group_range<D>(queue);
+  size_t work_group_size = work_group_range.size();
+
+  // array to return results
+  T res[test_matrix] = {init_helper<T>(0)};
+  {
+    sycl::buffer<T, 1> res_sycl(res, sycl::range<1>(test_matrix));
+
+    queue.submit([&](sycl::handler& cgh) {
+      auto res_acc =
+          res_sycl.template get_access<sycl::access::mode::read_write>(cgh);
+
+      sycl::nd_range<D> executionRange(work_group_range, work_group_range);
+
+      cgh.parallel_for<broadcast_group_kernel<D, T>>(
+          executionRange, [=](sycl::nd_item<D> item) {
+            sycl::group<D> group = item.get_group();
+
+            // find local id of last group item
+            sycl::id<D> last_item = group.get_local_range();
+            for (int i = 0; i < D; ++i) {
+              --last_item[i];
+            }
+
+            T local_var = init_helper<T>(item.get_local_linear_id() + 1);
```

Comment:
I have no idea about what could be a helper in an init context. :-(

Reply:
See your next comment. :)